src/builder.rs

```diff
@@ -92,12 +91,16 @@ impl StatsdBuilder {
         interval: Duration,
     ) -> Result<Self, BuildError>
     where
-        T: AsRef<str>,
+        T: ToSocketAddrs,
     {
+        let endpoint = endpoint
+            .to_socket_addrs()
+            .map_err(|e| BuildError::InvalidPushGatewayEndpoint(e.to_string()))?
+            .next() // just use the first address we resolve to
+            .unwrap();
```

Comment:
Don't use `unwrap()`, it panics if the value is `None`.

Instead, turn `None` into a error result:

```rust
        let endpoint = endpoint
            .to_socket_addrs()
            .map_err(|e| BuildError::InvalidPushGatewayEndpoint(e.to_string()))?
            .next() // just use the first address we resolve to
            .ok_or_else(|| BuildError::InvalidPushGatewayEndpoint("to_socket_addrs returned an empty iterator"))?;
```

Reply:
Good point